dtfs-central-api/src/services/state-machines/utilisation-report/event-handlers/add-fees-to-an-existing-payment-group/add-fees-to-an-existing-payment-group.event-handler.ts

```diff
@@ -0,0 +1,90 @@
+  feeRecordsToAdd: FeeRecordEntity[];
+  otherFeeRecordsInPaymentGroup: FeeRecordEntity[];
+  payments: PaymentEntity[];
+  requestSource: DbRequestSource;
+};
+
+export type UtilisationReportAddFeesToAnExistingPaymentGroupEvent = BaseUtilisationReportEvent<
+  'ADD_FEES_TO_AN_EXISTING_PAYMENT_GROUP',
+  AddFeesToAnExistingPaymentGroupEventPayload
+>;
+
+const addSelectedFeeRecordsToPaymentGroup = async (
+  transactionEntityManager: EntityManager,
+  feeRecordsToAdd: FeeRecordEntity[],
+  feeRecordsAndPaymentsMatch: boolean,
+  payments: PaymentEntity[],
+  requestSource: DbRequestSource,
+) => {
+  await Promise.all(
+    payments.map(async (payment) => {
+      payment.updateWithAdditionalFeeRecords({
+        feeRecords: feeRecordsToAdd,
+        requestSource,
+      });
+      await transactionEntityManager.save(PaymentEntity, payment);
+    }),
+  );
+
+  const feeRecordStateMachines = feeRecordsToAdd.map((feeRecord) => FeeRecordStateMachine.forFeeRecord(feeRecord));
+  await Promise.all(
+    feeRecordStateMachines.map((stateMachine) =>
+      stateMachine.handleEvent({
+        type: 'PAYMENT_ADDED',
+        payload: {
+          transactionEntityManager,
+          feeRecordsAndPaymentsMatch,
+          requestSource,
+        },
+      }),
+    ),
+  );
+};
+
+const updateOtherFeeRecordsInPaymentGroup = async (
+  transactionEntityManager: EntityManager,
+  feeRecordsToUpdate: FeeRecordEntity[],
+  feeRecordsAndPaymentsMatch: boolean,
+  requestSource: DbRequestSource,
+) => {
+  const feeRecordStateMachines = feeRecordsToUpdate.map((feeRecord) => FeeRecordStateMachine.forFeeRecord(feeRecord));
+  await Promise.all(
+    feeRecordStateMachines.map((stateMachine) =>
+      stateMachine.handleEvent({
+        type: 'OTHER_FEE_ADDED_TO_PAYMENT_GROUP',
+        payload: {
+          transactionEntityManager,
+          feeRecordsAndPaymentsMatch,
+          requestSource,
+        },
+      }),
+    ),
+  );
+};
+
+export const handleUtilisationReportAddFeesToAnExistingPaymentGroupEvent = async (
+  report: UtilisationReportEntity,
+  { transactionEntityManager, feeRecordsToAdd, otherFeeRecordsInPaymentGroup, payments, requestSource }: AddFeesToAnExistingPaymentGroupEventPayload,
+): Promise<UtilisationReportEntity> => {
+  const allFeeRecords = [...otherFeeRecordsInPaymentGroup, ...feeRecordsToAdd];
+  const feeRecordsAndPaymentsMatchAfterFeeRecordsAdded = await feeRecordsMatchAttachedPayments(allFeeRecords, transactionEntityManager);
+
+  await addSelectedFeeRecordsToPaymentGroup(transactionEntityManager, feeRecordsToAdd, feeRecordsAndPaymentsMatchAfterFeeRecordsAdded, payments, requestSource);
+  await updateOtherFeeRecordsInPaymentGroup(
+    transactionEntityManager,
+    otherFeeRecordsInPaymentGroup,
+    feeRecordsAndPaymentsMatchAfterFeeRecordsAdded,
+    requestSource,
+  );
+
+  report.updateLastUpdatedBy(requestSource);
+  return await transactionEntityManager.save(UtilisationReportEntity, report);
```

Comment:
I think the latest on the decision on what should count as an update to a report is that it should just be if a field on that entity itself has changed, so here the changes are not to the report entity itself so we should just return it without any updates.

Reply:
Ah that's nice, removed this now thanks